modules/honzon-distribution/src/tests.rs

```diff
@@ -0,0 +1,237 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2022 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Unit tests for Honzon distribution module.
+
+#![cfg(test)]
+
+use super::*;
+use crate::mock::{Event, *};
+use frame_support::{assert_noop, assert_ok};
+use nutsfinance_stable_asset::traits::StableAsset as StableAssetT;
+
+fn initial_stable_asset(currency_id_a: CurrencyId, currency_id_b: CurrencyId) -> DispatchResult {
+	let amount = 100_000_000_000_000u128;
+	StableAssetWrapper::create_pool(
+		STABLE_ASSET,
+		vec![currency_id_a, currency_id_b],
+		vec![1u128, 1u128],
+		0,
+		0,
+		0,
+		3000u128,
+		BOB,
+		BOB,
+		10_000_000_000u128,
+	)?;
+
+	Tokens::deposit(currency_id_a, &BOB, amount)?;
+	Tokens::deposit(currency_id_b, &BOB, amount)?;
+
+	StableAssetWrapper::mint(&BOB, 0, vec![amount, amount], 0)?;
+	assert_eq!(
+		StableAssetWrapper::pool(0).map(|p| p.balances).unwrap(),
+		vec![amount, amount]
+	);
+
+	Ok(())
+}
+
+#[test]
+fn update_params_works() {
+	ExtBuilder::default().build().execute_with(|| {
+		let distribution_to_stable_asset = DistributionToStableAsset::<AccountId> {
+			pool_id: 0,
+			stable_token_index: 0,
+			stable_currency_id: AUSD,
+			account_id: CHARLIE,
+		};
+		let destination = DistributionDestination::StableAsset(distribution_to_stable_asset);
+
+		assert_ok!(HonzonDistribution::update_params(
+			Origin::root(),
+			destination.clone(),
+			Some(1000),
+			None,
+			None,
+			None
+		));
+
+		System::assert_last_event(Event::HonzonDistribution(crate::Event::UpdateDistributionParams {
+			destination,
+			params: DistributionParams {
+				capacity: 1000,
+				max_step: 0,
+				target_min: Default::default(),
+				target_max: Default::default(),
+			},
+		}));
+	});
+}
+
+#[test]
+fn stable_asset_mint_burn_works() {
```

Comment:
Maybe add another test with three pool to make sure it works with that case

Reply:
I'll do that in integration test instead of here.